testing/trino-server-dev/etc/catalog/hive.properties

```diff
@@ -8,8 +8,9 @@
 connector.name=hive
 
 # Configuration appropriate for Hive as started by product test environment, e.g.
-#   trino-product-tests-launcher/bin/run-launcher env up --environment singlenode --without-trino
+#   testing/trino-product-tests-launcher/bin/run-launcher env up --environment singlenode-hdfs-impersonation --without-trino
 # On Mac, this additionally requires that you add "<your external IP> hadoop-master" to /etc/hosts
+# On Apple Silicon set TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK=true
```

Comment:
This means we'll run some services with emulation. Maybe it's a good moment to figure out how to rebuild them for arm64. Which services are required here?

Reply:
This is specifically for running hive metastore.

Reply:
This might not be a problem with the latest Docker: https://levelup.gitconnected.com/docker-on-apple-silicon-mac-how-to-run-x86-containers-with-rosetta-2-4a679913a0d5you

Reply:
`singlenode-hdfs-impersonation` includes the `Hadoop` env which, by default, uses the `ghcr.io/trinodb/testing/hdp2.6-hive` image. We won't be able to build this one for arm64, since we don't have the original Dockerfile anymore, and it's not easy to swap out the base image without it.

I'd suggest using any other env that uses the `hive3.1-hive` image, which is multi-arch. Here's a list of which images are single and which are multi-arch: https://github.com/trinodb/docker-images/blob/master/.github/workflows/release.yml#L71 

Unfortunately, looks like we currently don't run any tests with `hive3.1-hive`. It's used in the `ConfigApacheHive3` class and the usage of this config is commented out here: https://github.com/trinodb/trino/blob/master/.github/workflows/ci.yml#L257. But maybe it has a working HMS and that'll be enough for the local dev env? Can you give it a try?

Reply:
Yet another "benefit" of TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK. I think we should remove it and keep it enabled on CI only.
cc @wendigo @kokosing

Reply:
I would change 
```
 On Apple Silicon set TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK=true
```

to
```
It is not recommended on Apple Silicon. This environment might be unreliable and not interactive. Such configuration is prohibited by the check
```

I would not not mention the env variable as this is already printed when it is executed.

> Yet another "benefit" of TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK. I

Exactly. We just saved people time and they are not going to even waste time with Hive docker on M1.

Reply:
I think we should just remove the check. I doubt it's a time saver overall